mm2src/coins/utxo/rpc_clients.rs

```diff
@@ -1108,38 +1133,37 @@ fn electrum_connect(
             delay = 5;
             let stream_clone = try_loop!(stream.as_ref().try_clone(), rx, connection, delay);
             let last_chunk = Arc::new(AtomicU64::new(now_ms()));
-            let interval = Interval::new(Duration::from_secs(ELECTRUM_TIMEOUT)).map_err(|e| { log!([e]); () });
-            CORE.spawn({
-                let last_chunk = last_chunk.clone();
-                move |_| interval.for_each(move |_| {
-                    let last = last_chunk.load(AtomicOrdering::Relaxed);
-                    if now_ms() > last + ELECTRUM_TIMEOUT * 1000 {
+            let last_chunkʹ = last_chunk.clone();
+            spawn ((async move || {
+                loop {
+                    Timer::after(ELECTRUM_TIMEOUT as f64).await;
+                    let last = last_chunkʹ.load(AtomicOrdering::Relaxed);
+                    if now_ms() - last < ELECTRUM_TIMEOUT * 1000 {continue}
```

Comment:
Since now_ms() can move backwards and then [cause integer underflow panic](https://github.com/KomodoPlatform/atomicDEX-API/issues/315) I prefer to not subtract the previous time from `now` and convert the inequation to perform addition instead. Many things in MM2 can go wrong if system time changes significantly, but if it's just few minutes we might keep it running without any issue.
As already mentioned at #315 we should rethink the way we work with time, but it's topic for separate discussion.

Reply:
Right, thanks for the catch!
I hear unsigneds are often avoided for this very reason.
Timers are never precise anyway and `f64` allows us to represent time in seconds which I think is more readable. Time deltas are a bit more readable to me as well, it might be simpler to have the delta on one side of the comparison and the timeout value on the other.

Reply:
BTW, I'm using https://crates.io/crates/atomic now as it's WASM-compatible and it can probably represent `Atomic<f64>` as well.

Reply:
Yea, agree about delta. I've used addition approach and already had some problems with it understanding my own code :slightly_smiling_face: I will refactor these places to signed deltas later so we will have same approach everywhere.